app/models/embedded_ansible_worker/runner.rb

```diff
@@ -40,11 +39,10 @@ def setup_ansible
   end
 
   def do_work
-    if EmbeddedAnsible.running?
-      _log.info("#{log_prefix} supervisord is ok!")
+    if EmbeddedAnsible.alive?
+      heartbeat
```

Comment:
Doesn't this introduce a delay?  That is, you have to wait for the worker heartbeat detection to stop the worker.